pkg/server/stream/v3/stream.go

```diff
@@ -54,6 +58,7 @@ func NewStreamState(wildcard bool, initialResourceVersions map[string]string) St
 		first:                   true,
 		knownResourceNames:      map[string]map[string]struct{}{},
 		ordered:                 false, // Ordered comes from the first request since that's when we discover if they want ADS
+		hasWarmingCluster:       false,
```

Comment:
As @valerian-roche suggested in my PR [envoyproxy/go-control-plane#981](https://github.com/envoyproxy/go-control-plane/pull/981), it would be beneficial to push only the endpoints of updated clusters. When your Envoy instance is watching over 1000 clusters, forcing a full update for all of them can be excessive if only one has changed. I could reopen my PR and finish it up if it makes sense to do it. Envoy has a new feature to load cached EDS but it works ONLY when `initial_fetch_timeout` is not disabled.

Reply:
It seems to me, only 1 extra empty EDS response is needed no matter how many Cluster is updated.

So my patch will only set an empty EDS response, if nothing changes.